### DISCUSSION TITLE
Handling indents

### DISCUSSION THREAD

**Discussion by vdavez** [pymupdf/PyMuPDF#2197] [Looking for help]
Hello—First of all, this library is amazing. Thank you! 

I am working with a [PDF](https://www.govinfo.gov/content/pkg/CRPT-117hrpt507/PDF/CRPT-117hrpt507.pdf) where indentation matters to a human but the PDF itself doesn't include any whitespace characters in the document. When I convert to `xhtml` (my desired output), I lose the whitespace.

Is there any smart way to determine whether a line of text starts with whitespace and then insert whitespace characters so that when I output to xhtml, I can preserve that aspect of the layout?

Here's a screenshot of what I'm working with...

![117hrpt5076](https://user-images.githubusercontent.com/4153048/215153294-18b3c905-6117-4c0b-bf9b-21d7ab9fa318.png)

Thank you!!!

**Comment by JorjMcKie** [comment]
No the only way is to note at which x-coordinate the first character starts and set this into relation to page width and the corresponding values of other lines.
To do this your output format "xhtml" is unfortunately chosen, because it contains no position information ...

Why don't you use the "dict" format:
```python
for block in page.get_text("dict", flags=fitz.TEXTFLAGS_TEXT)["blocks"]:
    for line in block["lines"]:
        bbox = line["bbox"]
        text = "".join([span["text"] for span in line["spans"]])
        print(f"line '{text}' starts at {bbox[0]}")
```

To be a little picky with the wording:
You don't actually **_"lose"_** spaces: they never existed!
